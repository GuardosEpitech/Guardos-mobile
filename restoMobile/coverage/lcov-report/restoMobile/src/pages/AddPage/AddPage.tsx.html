
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for restoMobile/src/pages/AddPage/AddPage.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../../index.html">All files</a> / <a href="index.html">restoMobile/src/pages/AddPage</a> AddPage.tsx</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">68.75% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>22/32</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">33.33% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>2/6</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">50% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>5/10</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">70% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>21/30</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a>
<a name='L65'></a><a href='#L65'>65</a>
<a name='L66'></a><a href='#L66'>66</a>
<a name='L67'></a><a href='#L67'>67</a>
<a name='L68'></a><a href='#L68'>68</a>
<a name='L69'></a><a href='#L69'>69</a>
<a name='L70'></a><a href='#L70'>70</a>
<a name='L71'></a><a href='#L71'>71</a>
<a name='L72'></a><a href='#L72'>72</a>
<a name='L73'></a><a href='#L73'>73</a>
<a name='L74'></a><a href='#L74'>74</a>
<a name='L75'></a><a href='#L75'>75</a>
<a name='L76'></a><a href='#L76'>76</a>
<a name='L77'></a><a href='#L77'>77</a>
<a name='L78'></a><a href='#L78'>78</a>
<a name='L79'></a><a href='#L79'>79</a>
<a name='L80'></a><a href='#L80'>80</a>
<a name='L81'></a><a href='#L81'>81</a>
<a name='L82'></a><a href='#L82'>82</a>
<a name='L83'></a><a href='#L83'>83</a>
<a name='L84'></a><a href='#L84'>84</a>
<a name='L85'></a><a href='#L85'>85</a>
<a name='L86'></a><a href='#L86'>86</a>
<a name='L87'></a><a href='#L87'>87</a>
<a name='L88'></a><a href='#L88'>88</a>
<a name='L89'></a><a href='#L89'>89</a>
<a name='L90'></a><a href='#L90'>90</a>
<a name='L91'></a><a href='#L91'>91</a>
<a name='L92'></a><a href='#L92'>92</a>
<a name='L93'></a><a href='#L93'>93</a>
<a name='L94'></a><a href='#L94'>94</a>
<a name='L95'></a><a href='#L95'>95</a>
<a name='L96'></a><a href='#L96'>96</a>
<a name='L97'></a><a href='#L97'>97</a>
<a name='L98'></a><a href='#L98'>98</a>
<a name='L99'></a><a href='#L99'>99</a>
<a name='L100'></a><a href='#L100'>100</a>
<a name='L101'></a><a href='#L101'>101</a>
<a name='L102'></a><a href='#L102'>102</a>
<a name='L103'></a><a href='#L103'>103</a>
<a name='L104'></a><a href='#L104'>104</a>
<a name='L105'></a><a href='#L105'>105</a>
<a name='L106'></a><a href='#L106'>106</a>
<a name='L107'></a><a href='#L107'>107</a>
<a name='L108'></a><a href='#L108'>108</a>
<a name='L109'></a><a href='#L109'>109</a>
<a name='L110'></a><a href='#L110'>110</a>
<a name='L111'></a><a href='#L111'>111</a>
<a name='L112'></a><a href='#L112'>112</a>
<a name='L113'></a><a href='#L113'>113</a>
<a name='L114'></a><a href='#L114'>114</a>
<a name='L115'></a><a href='#L115'>115</a>
<a name='L116'></a><a href='#L116'>116</a>
<a name='L117'></a><a href='#L117'>117</a>
<a name='L118'></a><a href='#L118'>118</a>
<a name='L119'></a><a href='#L119'>119</a>
<a name='L120'></a><a href='#L120'>120</a>
<a name='L121'></a><a href='#L121'>121</a>
<a name='L122'></a><a href='#L122'>122</a>
<a name='L123'></a><a href='#L123'>123</a>
<a name='L124'></a><a href='#L124'>124</a>
<a name='L125'></a><a href='#L125'>125</a>
<a name='L126'></a><a href='#L126'>126</a>
<a name='L127'></a><a href='#L127'>127</a>
<a name='L128'></a><a href='#L128'>128</a>
<a name='L129'></a><a href='#L129'>129</a>
<a name='L130'></a><a href='#L130'>130</a>
<a name='L131'></a><a href='#L131'>131</a>
<a name='L132'></a><a href='#L132'>132</a>
<a name='L133'></a><a href='#L133'>133</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { useEffect, useCallback, useState } from "react";
import { Text, View, ScrollView, ImageBackground, TouchableOpacity } from "react-native";
import styles from "./AddPage.styles";
import Icon from "react-native-vector-icons/Entypo";
import Trash from "react-native-vector-icons/FontAwesome";
// @ts-ignore
import { API_URL } from '@env';
import * as SplashScreen from "expo-splash-screen";
import axios from "axios";
import IconBack from "react-native-vector-icons/AntDesign";
import IconUser from "react-native-vector-icons/FontAwesome";
import { IIngredients } from "src/models/ingedientsInterfaces";
&nbsp;
&nbsp;
const URL = API_URL + "/get";
&nbsp;
SplashScreen.preventAutoHideAsync();
&nbsp;
const AddPage = ({ navigation }: { navigation: any }) =&gt; {
  const [appIsReady, setAppIsReady] = useState(false);
&nbsp;
  useEffect(() =&gt; {
    async function prepare() {
      try {
        await SplashScreen.preventAutoHideAsync();
        await new Promise((resolve) =&gt; setTimeout(resolve, 2000));
      } catch (e) {
        console.warn(e);
      } finally {
        setAppIsReady(true);
      }
    }
&nbsp;
    prepare();
  }, []);
  const [IngrValue, setIngrValue] = useState&lt;[IIngredients]&gt;();
&nbsp;
  const onLayoutRootView = useCallback(<span class="fstat-no" title="function not covered" >as</span>ync () =&gt; {
<span class="cstat-no" title="statement not covered" >    if (appIsReady) {</span>
<span class="cstat-no" title="statement not covered" >      await SplashScreen.hideAsync();</span>
    }
  }, [appIsReady]);
&nbsp;
  if (!appIsReady) {
    return null;
  }
&nbsp;
  function <span class="fstat-no" title="function not covered" >GoToQRCode(</span>) {
<span class="cstat-no" title="statement not covered" >    navigation.navigate("QRCodeEngin");</span>
  }
&nbsp;
  async function getIngr() {
    try {
      const response = await axios({
        method: "get",
        url: API_URL + "/get",
      });
      setIngrValue(response.data);
    } catch (err) {
      console.log(err);
    }
  }
  async function <span class="fstat-no" title="function not covered" >deleteIngr(</span>body: string) {
<span class="cstat-no" title="statement not covered" >    try {</span>
      const response = <span class="cstat-no" title="statement not covered" >await axios({</span>
        method: "delete",
        url: API_URL + "/delete",
        data: { name: body },
      });
<span class="cstat-no" title="statement not covered" >      if (response.status == 200) <span class="cstat-no" title="statement not covered" >console.log("DELETE");</span></span>
    } catch (err) {
<span class="cstat-no" title="statement not covered" >      console.log(err);</span>
    }
  }
&nbsp;
  getIngr();
&nbsp;
  return (
    &lt;View style={{ flex: 1 }}&gt;
      &lt;ImageBackground source={require('../../assets/background.png')} resizeMode="cover" style={styles.image}&gt;
        &lt;View onLayout={onLayoutRootView} style={{ alignItems: "center" }}&gt;
          &lt;View style={styles.DivTop}&gt;
            &lt;IconBack
              style={styles.IconBack}
              name="left"
              size={40}
              color="#4D4D4D"
            /&gt;
            &lt;Text style={styles.CategorieTitle}&gt;Ingredients&lt;/Text&gt;
            &lt;IconUser
              style={styles.IconUser}
              name="user"
              size={40}
              color="#4D4D4D"
            /&gt;
          &lt;/View&gt;
          &lt;View style={styles.DivTitleIngr}&gt;
            &lt;Text style={styles.TitleIngr}&gt;My Ingredients&lt;/Text&gt;
          &lt;/View&gt;
          &lt;ScrollView style={styles.ScrollView}&gt;
            {IngrValue?.map(<span class="fstat-no" title="function not covered" >(i</span>ngr) =&gt; (
<span class="cstat-no" title="statement not covered" >              &lt;View style={{ alignItems: "center" }} key={ingr.name}&gt;</span>
                &lt;View style={styles.RectIngr} key={ingr.name}&gt;
                  &lt;Text style={styles.TitleIngrList}&gt;{ingr.name}&lt;/Text&gt;
                  &lt;Trash
                    style={styles.TrashIcon}
                    name="trash"
                    size={30}
                    color="#6D071A"
                    onPress={<span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >                      deleteIngr(ingr.name);</span>
                    }}
                  /&gt;
                &lt;/View&gt;
              &lt;/View&gt;
            ))}
          &lt;/ScrollView&gt;
          &lt;TouchableOpacity style={styles.ViewIcon} onPress={GoToQRCode}&gt;
            &lt;Icon
              name="plus"
              size={30}
              color="#fff"
            /&gt;
            &lt;Text style={styles.TitleAddIngr}&gt;Add Ingredients&lt;/Text&gt;
          &lt;/TouchableOpacity&gt;
        &lt;/View&gt;
      &lt;/ImageBackground&gt;
    &lt;/View&gt;
  );
};
&nbsp;
export default AddPage;
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2024-05-28T20:07:29.542Z
            </div>
        <script src="../../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../../sorter.js"></script>
        <script src="../../../../block-navigation.js"></script>
    </body>
</html>
    