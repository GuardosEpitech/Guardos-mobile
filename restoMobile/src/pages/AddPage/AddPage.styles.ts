import { StyleSheet, Dimensions, Platform } from "react-native";

const gap = 90;
const deviceDisplay = Dimensions.get("window");
const deviceWidth = deviceDisplay.width;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
  },
  DivTop: {
    marginTop: Platform.OS === 'ios' ? 0 : 35,
    flexDirection: 'row',
    alignItems: 'center',
    paddingRight: 10,
    paddingHorizontal: (gap / -2),
  },
  DivTitleIngr: {
    marginTop: 10,
  },
  TitleIngr: {
    fontFamily: "Montserrat_700Bold",
    fontSize: 32,
    lineHeight: 39,
    textAlign: "center",
    color: "#56885E",
  },
  TitleIngrList: {
    fontFamily: "Montserrat_700Bold",
    fontSize: 24,
    color: "#646464",
    marginLeft: 10,
    alignSelf: "flex-start",
    position: "absolute",
  },
  TrashIcon: {
    alignSelf: "flex-end",
    marginRight: 10,
  },
  ImgLogo: {
    width: 105,
    height: 105,
  },
  ViewIcon: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    width: 390,
    height: 66,
    backgroundColor: "#56885E",
    borderRadius: 10,
  },

  RectIngr: {
    shadowColor: 'black',
    shadowOpacity: 0.25,
    shadowOffset: { width: 0, height: 4},
    shadowRadius: 10,
    elevation: 1,
    backgroundColor: 'white',
    width: 370,
    height: 80,
    margin: 10,
    borderRadius: 5,
    display: "flex",
    justifyContent: "center",
  },
  ScrollView: {
    marginTop: 30,
    height: 600,
    width: deviceWidth,
    paddingTop: 10,
  },
  TitleAddIngr: {
    color: "white",
    fontFamily: "Montserrat_700Bold",
    fontSize: 23,
  },
  IconUser: {
    marginHorizontal: gap / 2,
  },
  IconBack: {
    marginHorizontal: gap / 2,
  },
  CategorieTitle: {
    color: "#4D4D4D",
    fontFamily: "Montserrat_700Bold",
    fontSize: 23,
    marginHorizontal: gap / 2,
  },
  image: {
    flex: 1,
    justifyContent: "center"
  },
});

export default styles;
