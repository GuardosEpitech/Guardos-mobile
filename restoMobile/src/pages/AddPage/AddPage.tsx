import React, { useEffect, useCallback, useState } from "react";
import { Text, View, ScrollView, ImageBackground, TouchableOpacity } from "react-native";
import styles from "./AddPage.styles";
import Icon from "react-native-vector-icons/Entypo";
import Trash from "react-native-vector-icons/FontAwesome";
import * as SplashScreen from "expo-splash-screen";
import axios from "axios";
import { API_URL } from '@env';
import IconBack from "react-native-vector-icons/AntDesign";
import IconUser from "react-native-vector-icons/FontAwesome";
import { IIngredients } from "src/models/ingedientsInterfaces";

console.log(API_URL)

SplashScreen.preventAutoHideAsync(); // Keep the splash screen visible while we fetch resources

import {
  useFonts,
  Montserrat_100Thin,
  Montserrat_300Light,
  Montserrat_400Regular,
  Montserrat_500Medium,
  Montserrat_600SemiBold,
  Montserrat_700Bold,
} from "@expo-google-fonts/montserrat";



const AddPage = ({ navigation }: { navigation: any }) => {
  let [fontsLoaded] = useFonts({
    Montserrat_100Thin,
    Montserrat_300Light,
    Montserrat_400Regular,
    Montserrat_500Medium,
    Montserrat_600SemiBold,
    Montserrat_700Bold,

  });
  const [appIsReady, setAppIsReady] = useState(false);

  useEffect(() => {
    async function prepare() {
      try {
        await SplashScreen.preventAutoHideAsync();
        await fontsLoaded;
        await new Promise((resolve) => setTimeout(resolve, 2000));
      } catch (e) {
        console.warn(e);
      } finally {
        setAppIsReady(true);
      }
    }

    prepare();
  }, []);
  const [IngrValue, setIngrValue] = useState<[IIngredients]>();

  const onLayoutRootView = useCallback(async () => {
    if (appIsReady) {
      await SplashScreen.hideAsync();
    }
  }, [appIsReady]);

  if (!appIsReady) {
    return null;
  }

  function GoToQRCode() {
    navigation.navigate("QRCodeEngin");
  }

  async function getIngr() {
    try {
      const response = await axios({
        method: "get",
        url: API_URL + "/get",
      });
      setIngrValue(response.data);
    } catch (err) {
      console.log(err);
    }
  }
  async function deleteIngr(body: string) {
    try {
      const response = await axios({
        method: "delete",
        url: API_URL + "/delete",
        data: { name: body },
      });
      if (response.status == 200) console.log("DELETE");
    } catch (err) {
      console.log(err);
    }
  }

  getIngr();

  return (
    <View style={{ flex: 1 }}>
      <ImageBackground source={require('../../assets/background.png')} resizeMode="cover" style={styles.image}>
        <View onLayout={onLayoutRootView} style={{ alignItems: "center" }}>
          <View style={styles.DivTop}>
            <IconBack
              style={styles.IconBack}
              name="left"
              size={40}
              color="#4D4D4D"
            />
            <Text style={styles.CategorieTitle}>Ingredients</Text>
            <IconUser
              style={styles.IconUser}
              name="user"
              size={40}
              color="#4D4D4D"
            />
          </View>
          <View style={styles.DivTitleIngr}>
            <Text style={styles.TitleIngr}>My Ingredients</Text>
          </View>
          <ScrollView style={styles.ScrollView}>
            {IngrValue?.map((ingr) => (
              <View style={{ alignItems: "center" }} key={ingr.name}>
                <View style={styles.RectIngr} key={ingr.name}>
                  <Text style={styles.TitleIngrList}>{ingr.name}</Text>
                  <Trash
                    style={styles.TrashIcon}
                    name="trash"
                    size={30}
                    color="#6D071A"
                    onPress={() => {
                      deleteIngr(ingr.name);
                    }}
                  />
                </View>
              </View>
            ))}
          </ScrollView>
          <TouchableOpacity style={styles.ViewIcon} onPress={GoToQRCode}>
            <Icon
              name="plus"
              size={30}
              color="#fff"
            />
            <Text style={styles.TitleAddIngr}>Add Ingredients</Text>
          </TouchableOpacity>
        </View>
      </ImageBackground>
    </View>
  );
};

export default AddPage;
