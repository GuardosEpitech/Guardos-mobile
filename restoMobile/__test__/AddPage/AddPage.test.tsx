import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import AddPage from 'src/pages/AddPage/AddPage';
import axios from 'axios';
import * as SplashScreen from 'expo-splash-screen';

jest.mock('axios');
jest.mock('expo-splash-screen', () => ({
  preventAutoHideAsync: jest.fn(),
  hideAsync: jest.fn(),
}));

jest.mock('react-native-vector-icons/Entypo', () => 'Icon');
jest.mock('react-native-vector-icons/FontAwesome', () => 'Trash');
jest.mock('react-native-vector-icons/AntDesign', () => 'IconBack');
jest.mock('react-native-vector-icons/FontAwesome', () => 'IconUser');

describe('AddPage', () => {
  const mockNavigation = { navigate: jest.fn() };

  beforeEach(() => {
    jest.clearAllMocks();
    SplashScreen.preventAutoHideAsync.mockResolvedValue();
    SplashScreen.hideAsync.mockResolvedValue();
  });

  it('renders correctly and hides splash screen', async () => {
    const { getByText } = render(<AddPage navigation={mockNavigation} />);

    await waitFor(() => {
      expect(SplashScreen.hideAsync).toHaveBeenCalled();
    });

    expect(getByText('Ingredients')).toBeTruthy();
    expect(getByText('My Ingredients')).toBeTruthy();
    expect(getByText('Add Ingredients')).toBeTruthy();
  });

  it('fetches and displays ingredients', async () => {
    const mockIngredients = [{ name: 'Sugar' }, { name: 'Salt' }];
    axios.get.mockResolvedValue({ data: mockIngredients });

    const { getByText } = render(<AddPage navigation={mockNavigation} />);

    await waitFor(() => {
      expect(getByText('Sugar')).toBeTruthy();
      expect(getByText('Salt')).toBeTruthy();
    });
  });

  it('handles ingredient fetch error', async () => {
    axios.get.mockRejectedValue(new Error('Failed to fetch'));

    const { queryByText } = render(<AddPage navigation={mockNavigation} />);

    await waitFor(() => {
      expect(queryByText('Sugar')).toBeNull();
      expect(queryByText('Salt')).toBeNull();
    });
  });

  it('deletes an ingredient', async () => {
    const mockIngredients = [{ name: 'Sugar' }, { name: 'Salt' }];
    axios.get.mockResolvedValue({ data: mockIngredients });
    axios.delete.mockResolvedValue({ status: 200 });

    const { getByText, queryByText, getAllByRole } = render(<AddPage navigation={mockNavigation} />);

    await waitFor(() => {
      expect(getByText('Sugar')).toBeTruthy();
      expect(getByText('Salt')).toBeTruthy();
    });

    const deleteButtons = getAllByRole('button', { name: /trash/i });
    fireEvent.press(deleteButtons[0]);

    await waitFor(() => {
      expect(queryByText('Sugar')).toBeNull();
    });

    expect(axios.delete).toHaveBeenCalledWith('https://example.com/api/delete', { data: { name: 'Sugar' } });
  });

  it('handles ingredient delete error', async () => {
    const mockIngredients = [{ name: 'Sugar' }, { name: 'Salt' }];
    axios.get.mockResolvedValue({ data: mockIngredients });
    axios.delete.mockRejectedValue(new Error('Failed to delete'));

    const { getByText, getAllByRole } = render(<AddPage navigation={mockNavigation} />);

    await waitFor(() => {
      expect(getByText('Sugar')).toBeTruthy();
      expect(getByText('Salt')).toBeTruthy();
    });

    const deleteButtons = getAllByRole('button', { name: /trash/i });
    fireEvent.press(deleteButtons[0]);

    await waitFor(() => {
      expect(getByText('Sugar')).toBeTruthy();
    });

    expect(axios.delete).toHaveBeenCalledWith('https://example.com/api/delete', { data: { name: 'Sugar' } });
  });

  it('navigates to QRCodeEngin screen', () => {
    const { getByText } = render(<AddPage navigation={mockNavigation} />);

    fireEvent.press(getByText('Add Ingredients'));

    expect(mockNavigation.navigate).toHaveBeenCalledWith('QRCodeEngin');
  });

  it('handles SplashScreen error', async () => {
    SplashScreen.preventAutoHideAsync.mockRejectedValue(new Error('Failed to prevent auto hide'));

    const { getByText } = render(<AddPage navigation={mockNavigation} />);

    await waitFor(() => {
      expect(SplashScreen.hideAsync).toHaveBeenCalled();
    });

    expect(getByText('Ingredients')).toBeTruthy();
    expect(getByText('My Ingredients')).toBeTruthy();
    expect(getByText('Add Ingredients')).toBeTruthy();
  });
});
