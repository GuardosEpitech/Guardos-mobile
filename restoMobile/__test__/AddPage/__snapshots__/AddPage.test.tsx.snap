// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddPage /> renders correctly 1`] = `null`;
